src/Microsoft.AspNetCore.Razor.Design/build/netstandard2.0/Microsoft.AspNetCore.Razor.Design.CodeGeneration.targets

```diff
@@ -83,14 +82,22 @@
     </RazorTagHelper>
   </Target>
 
-  <Target Name="_ResolveRazorGenerateOutputs" AfterTargets="PrepareForRazorGenerate">
+  <Target Name="_ResolveRazorGenerateOutputs">
     <ItemGroup>
       <_RazorGenerateOutput Include="%(RazorGenerate.GeneratedOutput)" Condition="'%(RazorGenerate.GeneratedOutput)'!=''"/>    
     </ItemGroup>
   </Target>
 
+  <PropertyGroup>
+    <RazorCoreGenerateDependsOn>
+      _HashRazorGenerateInputs;
```

Comment:
Made these two details of `RazorCoreGenerate` since they do not produce any global state that's useful in the SDK.

Reply:
I think this is fine. Since we're crossing a component boundary it's probably good that RazorCoreGenerate has a depends on.

Reply:
OK I see now that you're doing this everywhere. I just made the opposite choice when building all of this except for the 'top level' things. We should talk about this and make a decision.